src/main/java/GameMonitor.java

```diff
@@ -4,30 +4,53 @@
     Players secondPlayer;
     Game match;
 
-    public GameMonitor () {
+    public GameMonitor() {
         this.firstPlayer = new Players();
         this.secondPlayer = new Players();
         this.match = new Game(firstPlayer, secondPlayer);
     }
 
     public void buttonPress(Message message) {
+        if (message.getButtonId().equals("1") && match.whichPlayerSide(firstPlayer, secondPlayer)) {
```

Comment:
We can encapsulate the logic of calculating which player has pressed the button into a separate method.